src/grid.rs

```diff
@@ -0,0 +1,269 @@
+        };
+
+        count
+    }
+
+    fn clock(&mut self) {
+        let mut new_matrix = self.matrix.clone();
+
+        for (x, row) in new_matrix.iter_mut().enumerate().take(self.width) {
+            for (y, field) in row.iter_mut().enumerate().take(self.width) {
+                let neighbours_ammount = self.count_neighbours(x, y);
+                if ((neighbours_ammount == 2 || neighbours_ammount == 3) && *field == 1) ||
+                    (neighbours_ammount == 3 && *field == 0)
+                {
+                    *field = 1;
+                } else {
+                    *field = 0;
+                }
+            }
+        }
+
+        self.matrix = new_matrix;
+    }
+
+    fn print(&self) {
+        print!("  |");
+        for x in 0..self.width {
+            print!("{} ", x);
+        }
+        println!();
+        println!("--|--------------------");
+
+        for x in 0..self.width {
+            print!("{} |", x);
+            for y in 0..self.height {
+                if self.matrix[x][y] == 1 {
+                    print!("X ")
+                } else {
+                    print!("  ")
+                }
+                //print!(" {} ", self.matrix[x][y]);
+            }
+            println!();
+        }
+        println!();
+    }
+
+    pub fn start(&mut self) {
+        let mut generation = 0;
+
+        loop {
+            println!("Generation N˚ {}\n", generation);
+            self.print();
+            self.clock();
+            generation += 1;
+            sleep(Duration::from_secs(3));
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::Grid;
+
+    #[test]
+    fn test_01_new_returns_a_valid_matrix() {
+        // Create a matrix and it begins initialized with the right ammount of columns and
+        // rows, with al the fields in 0
+        let grid = Grid::new(10, 10);
+        assert_eq!(grid.height, grid.matrix.len());
+        assert_eq!(grid.width, grid.matrix.get(0).unwrap().len());
+        assert_eq!(0, grid.matrix[0][0]);
+    }
+
+    #[test]
+    fn test_02_seed_makes_live_passed_points() {
+        // As seed I can give the grid a vector of points, and those points are set in 1
+        let mut grid = Grid::new(10, 10);
+        let points = vec![(4, 4), (3, 4)];
+        grid.seed(points);
+
+        assert_eq!(1, grid.matrix[4][4]);
+        assert_eq!(1, grid.matrix[3][4]);
+    }
+
+    #[test]
+    fn test_03_clock_makes_die_cells_with_underpopulation() {
+        // When making a clock, the points that don't have at least 2 neighbours die
+        let mut grid = Grid::new(10, 10);
+        let points = vec![(4, 4), (3, 4)];
+        grid.seed(points);
+
+        assert_eq!(1, grid.matrix[4][4]);
+        assert_eq!(1, grid.matrix[3][4]);
+
+        grid.clock();
+
+        assert_eq!(0, grid.matrix[4][4]);
+        assert_eq!(0, grid.matrix[3][4]);
+    }
+
+    #[test]
+    fn test_04_clock_doesnt_make_die_cells_without_underpopulation_or_overpopulation() {
```

Comment:
```suggestion
    fn clock_doesnt_kill_cells_without_underpopulation_or_overpopulation() {
```
This test (along with some previous ones) could have a better name. The phrase ‘Doesn’t make die cells’ sounds awkward. I’ve provided a suggestion, but feel free to rename it as you see fit.